### PR TITLE
Add pre-commit and reformat the code

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
Using the same pre-config setup (black, prettier, flake8, mypy, rebase-check...) as we use in https://github.com/packit org.

@lslebodn Are you working on something? (The rebase would be really painful so I want to check before mergning.)

I'll update the contribution guide later.

### CODE REVIEW

File: colin/core/check_runner.py [hunk 1]
```diff
@@ -41,13 +39,13 @@ def _result_generator(target, checks, timeout=None):
                 logger.debug("Check timeout: {}".format(_timeout))
                 yield exit_after(_timeout)(check.check)(target)
             except TimeoutError as ex:
-                logger.warning(
-                    "The check hit the timeout: {}".format(_timeout))
+                logger.warning("The check hit the timeout: {}".format(_timeout))
```
Comment (lslebodn) [inline colin/core/check_runner.py:42]: I assume it was automatically re-formated but 
`string.format` should not be use in logger :-)

And there are few other similar places.

File: colin/core/checks/abstract_check.py [hunk 2]
```diff
@@ -33,15 +33,19 @@ def check(self, target):
...
+            "{}\n"
+            "   -> {}\n"
+            "   -> {}\n"
+            "   -> {}\n"
+            "   -> {}\n".format(
+                self.name,
+                self.message,
+                self.description,
+                self.reference_url,
+                ", ".join(self.tags),
```
Comment (lslebodn) [inline colin/core/checks/abstract_check.py:47]: We already depends on python-3.6

Should we use f-string?

It can be done in different PR

**Comment by lslebodn** [comment]
> @lslebodn Are you working on something? (The rebase would be really painful so I want to check before mergning.)

No, I plan to drop support for ostree (cause we already can use oci)
but I can do that after merging this PR :-)

File: colin/core/checks/abstract_check.py (hunk 2 above)
Comment (lachmanfrantisek) [reply colin/core/checks/abstract_check.py:47]: Nice! I'll create a followup for fstrings, format...

Comment (lslebodn) [reply colin/core/checks/abstract_check.py:47]: BTW we needn't use f-string everywhere but IMHO it would simplify some cases.

**Review by jpopelka — COMMENTED** [review]
I just skimmed through (since we haven't had issues with this pre-commit configuration in packit) and haven't seen anything unusual.

**Comment by lachmanfrantisek** [comment]
@lslebodn I have the follow-up PR prepared, can we merge this?

Can anyone, please, approve? Thanks!

**Review by lslebodn — APPROVED** [review]
test passed 
+1

Agree with additional changes in different PR

**Pull request merged**